CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -181,6 +197,33 @@ from UserCandies
             }
         }
 
+        // ConsumeRandomCandy: finds UserCandyId for oldest piece of a random candy by flavor and calls EatCandy
+        public UserCandyDetailed ConsumeRandomCandy(string flavor, int userId)
+        {
+            var sql = @"select top 1 with ties UserCandyId
+                        from UserCandies uc
+                            join Candies c
+                            on uc.candyId = c.candyId
+                        where isConsumed = 0
+                        and flavorCategory = @Flavor
+                        and userId = @UserId
+                        order by dateReceived";
+
+            using (var db = new SqlConnection(connectionString))
+            {
+                var parameters = new
+                {
+                    Flavor = flavor,
+                    UserId = userId
+                };
+                var CandyOptions = db.Query<int>(sql, parameters);
```

Comment:
This should have Query for a group of IEnumerables so you can select the index.  Replace "int" with <UserCandyDetailed>

Reply:
.Query<> already returns an IEnumerable, in this case, a collection of integers rather than a collection of class instances as we used elsewhere, because we only need the UserCandyId from this query.

Reply:
I was only able to return a single result because at this time you can only put one candy in at a time and the query does not filter down by day but all the way down to seconds....do we want this or do we want to only look at the day for filtering?

Reply:
good call on the time, i was testing with entries that had the same date and time and didn't think about that.  Updated to order by date without time